src/main/scala/org/ergoplatform/ErgoLikeTransaction.scala

```diff
@@ -113,49 +124,40 @@ object ErgoLikeTransaction {
     def bytesToSign(tx: ErgoLikeTransaction): Array[Byte] =
       bytesToSign(tx.inputs.map(_.boxId), tx.outputCandidates)
 
-
-    override def toBytes(ftx: FlattenedTransaction): Array[Byte] = {
-      ftx._1.map(_.spendingProof).foldLeft(bytesToSign(ftx._1.map(_.boxId), ftx._2)) { case (bytes, proof) =>
-        bytes ++ ProverResult.serializer.toBytes(proof)
+    override def serializeBody(ftx: FlattenedTransaction, w: ByteWriter): Unit = {
+      w.putBytes(bytesToSign(ftx.inputs.map(_.boxId), ftx.outputCandidates))
+      ftx.inputs.map(_.spendingProof).foreach { proof =>
+        ProverResult.serializer.serializeBody(proof, w)
       }
     }
 
-    override def parseBody(bytes: Array[Byte], pos: Position): (FlattenedTransaction, Consumed) = {
-      val r = Serializer.startReader(bytes, pos)
-      val inputsCount = r.getShort()
-
+    override def parseBody(r: ByteReader): FlattenedTransaction = {
+      val inputsCount = r.getUShort()
       val inputs = (0 until inputsCount).map { _ =>
         ADKey @@ r.getBytes(BoxId.size)
       }
-
-      val outsCount = r.getShort()
+      val outsCount = r.getUShort()
       val outputs = (0 until outsCount).map { _ =>
-        val (bc, cs) = ErgoBoxCandidate.serializer.parseBody(bytes, r.position)
-        r.position_=(r.position + cs)
-        bc
+        ErgoBoxCandidate.serializer.parseBody(r)
       }
 
       val proofs = (0 until inputsCount).map { _ =>
-        val (pr, cs) = ProverResult.serializer.parseBody(bytes, r.position)
-        r.position_=(r.position + cs)
-        pr
+        ProverResult.serializer.parseBody(r)
       }
 
       val signedInputs = inputs.zip(proofs).map { case (inp, pr) =>
         Input(inp, pr)
       }
-
-      (signedInputs, outputs) -> r.consumed
+      FlattenedTransaction(signedInputs, outputs)
```

Comment:
@greenhat please make sure `signedInputs` and `outputs` are internally represented by arrays, not lists

Reply:
Done https://github.com/ScorexFoundation/sigmastate-interpreter/pull/191/commits/57b49ba3822c1581b5b2feeb3a6ff72dda8757c9